pkg/provisioner/crdprovisioner.go

```diff
@@ -133,16 +134,35 @@ func (c *CrdProvisioner) CreateVolume(
 	secrets map[string]string,
 	volumeContentSource *v1alpha1.ContentVolumeSource,
 	accessibilityReq *v1alpha1.TopologyRequirement) (*v1alpha1.AzVolumeStatusParams, error) {
+
+	lister := c.conditionWatcher.informerFactory.Disk().V1alpha1().AzVolumes().Lister().AzVolumes(c.namespace)
+	azVolumeClient := c.azDiskClient.DiskV1alpha1().AzVolumes(c.namespace)
+
 	_, maxMountReplicaCount := azureutils.GetMaxSharesAndMaxMountReplicaCount(parameters)
 
 	// Getting the validVolumeName here since after volume
 	// creation the diskURI will consist of the validVolumeName
-	validVolumeName := azureutils.GetValidDiskName(volumeName)
-	azVolumeName := strings.ToLower(validVolumeName)
+	volumeName = azureutils.GetValidDiskName(volumeName)
+	azVolumeName := strings.ToLower(volumeName)
 
-	azV := c.azDiskClient.DiskV1alpha1().AzVolumes(c.namespace)
+	waiter, err := c.conditionWatcher.waitForObjectCondition(ctx, azvolume, azVolumeName, func(obj interface{}) (bool, error) {
```

Comment:
Consider `watchForObjectCondition` or `newObjectConditionWaiter` since the wait is performed elsewhere.